kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue

```diff
@@ -606,15 +597,17 @@
       this.translator = crossComponentTranslator(LibraryPage);
       this.filterTranslator = crossComponentTranslator(FilterTextbox);
       window.addEventListener('scroll', this.throttledHandleScroll);
-      this.setSearchWithinDescendant(this.topic);
-      this.search();
+      const loadingWatch = this.$watch('loading', () => {
+        this.setSearchWithinDescendant(this.topic);
+        this.search();
+        loadingWatch();
+      });
```

Comment:
This whole thing works so weirdly that `$watch` returns a function that stops the watcher. This isn't super common and maybe a comment or calling the variable `stopWatchingThis` would help clarify. Also, since these functions aren't async and you're not waiting for something to resolve before stopping the watcher, it might be clearer if you just declare the watcher into it's `stopMe` function, then call that stopper function outside of it's watcher handler's scope.

I may be reading this wrongly - so any clarification would be great (here or in code comments)